2.4/blocks/aspirelists/block_aspirelists.php

```diff
@@ -3,44 +3,93 @@
 // Released under the LGPL Licence - http://www.gnu.org/licenses/lgpl.html. Anyone is free to change or redistribute this code.
 
 class block_aspirelists extends block_base {
-  function init() {
-    $this->title   = get_config('aspirelists', 'blockTitle');
-  }
+    function init() {
+        $this->title     = get_config('aspirelists', 'blockTitle');
+    }
 
-  function get_content() {
-	global $CFG;
-	global $COURSE;
+    function get_content() {
+        global $CFG;
+        global $COURSE;
 
-    if ($this->content !== NULL) {
-      return $this->content;
-    }
+        if ($this->content !== NULL) {
+            return $this->content;
+        }
+
+        $hrefTarget = get_config('aspirelists', 'openNewWindow');
+        $target ='_self';
+        if($hrefTarget == 1){
+            $target = '_blank';
+        }
+
+        $this->content =    new stdClass;
+
+        if ($COURSE->idnumber)
+        {
+            $cache = cache::make('block_aspirelists', 'aspirelists');
+            $lists = $cache->get('list-' . $COURSE->idnumber);
+
+            if(!$lists) {
+                // get the code from the global course object
+                $lists = $this->getTalisAspireList($COURSE->idnumber);
+                $cache->set('list-' . $COURSE->idnumber, $lists);
```

Comment:
This ``list-`` prefix is used in a couple of places.  It would be good to turn this into a class constant at the top of this class, so as to lessen the chance of future bugs from typos.

Reply:
Done - I've defined a class constant for the `list-` prefix.